src/coreclr/jit/llvm.cpp

```diff
@@ -1382,6 +1382,31 @@ void Llvm::buildCnsInt(GenTree* node)
     failFunctionCompilation();
 }
 
+void Llvm::buildCnsLng(GenTree* node)
+{
+    if (node->gtType == TYP_LONG)
+    {
+        mapGenTreeToValue(node, _builder.getInt64(node->AsLngCon()->LngValue()));
+        return;
+    }
+    // TODO-LLVM: TYP_REF does not occur, but will it for Wasm64?
+    if (node->gtType == TYP_REF)
+    {
+        ssize_t longCon = node->AsLngCon()->LngValue();
+        // TODO: delete this check, just null ptr stores for now, other TYP_REFs not
+        // implemented yet
+        if (longCon != 0)
+        {
+            failFunctionCompilation();
+        }
+
+        mapGenTreeToValue(node, _builder.CreateIntToPtr(_builder.getInt64(longCon),
+                                                        Type::getInt8PtrTy(_llvmContext))); // TODO: wasm64
+        return;
+    }
+    failFunctionCompilation();
```

Comment:
```suggestion
    mapGenTreeToValue(node, _builder.getInt64(node->AsLngCon()->LngValue()));
```
`CNS_LNG` is only used for `TYP_LONG` on 32 bit targets.